File: AnySqlWebAdmin/Code/Math/cPoint.cs

using Microsoft.AspNetCore.Builder;
using Microsoft.CodeAnalysis.CSharp.Syntax;

namespace Vectors
{


    public class ParametricForms
    {
        
        // https://www.mathopenref.com/coordparamcircle.html
        public static void Circle()
        {
            double r = 20;
            double t = 33; // 0-2pi radian
            
            // x² + y² = r²
            // sin² + cos² = 1
            
            // x²/r² + y²/r² = 1
            
            
            double x = r * System.Math.Cos(t);
            double y = r * System.Math.Sin(t);
        }
        
        
        // https://math.stackexchange.com/questions/22064/calculating-a-point-that-lies-on-an-ellipse-given-an-angle
        public static void Ellipse()
        {
            //  radius along the x-axis is usually called a.
            double a = 30;
            //  radius along the y-axis is usually called b.
            double b = 15;
            double t = 33; // 0-2pi radian

            // Centered at the origin:
            
            // x = a cos t 
            // y = b sin t
            
            // x²/a² + y²/b² = 1
            // y = x*tan(t)
            
            // x²/a² + x²*tan²(t)/b² = 1
            // ==> x = +/- a*b/√(b²+a²*tan²(t))
            // where sign is + if -pi/2 < t < pi/2
            
            double h = 200; // x-coordinate of the ellipsis center
            double k = 200; // y-coordinate of the ellipsis center
            
            // in a circle, a = b = r 
            
            // if the circle is based on the origin, h&k = 0
            double x = h + a * System.Math.Cos(t);
            double y = k + b * System.Math.Sin(t);
            
        }


    }



    public class Point
    {
        public Point(double x, double y)
        {
            this.x = x;
            this.y = y;
        }

        public Point()
            :this(0,0)
        { }

        public double x;
        public double y;
    }

    public class cPoint
        : Point
    {
        public bool bCurrentlyValid = true;
        public bool bHasInterSection = false;
        public double z = 0;


        //Constructor
        public cPoint(double nXparam = 0, double nYparam = 0, double nZparam = 0)
        {
            this.x = nXparam;
            this.y = nYparam;
            this.z = nZparam;
            this.bCurrentlyValid = true;
        }  // End Constructor


        // instance.ToNormalPoint()
        public Point ToNormalPoint()
        {

            if (this.bCurrentlyValid)
            {
                Point ptReturnValue = new Point();
                ptReturnValue.x = this.x;
                ptReturnValue.y = this.y;
                return ptReturnValue;
            }

            return null;
        }


        // instance.toString();
        public override string ToString()
        {
            string strReturnValue = "(" + this.x + "," + this.y;
            if (z != 0)
                strReturnValue += ", " + this.z + ")";
            else
                strReturnValue += ")";

            if (!this.bCurrentlyValid)
                strReturnValue += " INVALID";

            return strReturnValue;
        }  // End function toString


    } // End cPoint 

} // End Package

File: AnySqlWebAdmin/wwwroot/ts/Math/cPoint.js
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var Vectors;
(function (Vectors) {
    var Point = (function () {
        function Point(nXparam, nYparam) {
            if (nXparam === void 0) { nXparam = 0; }
            if (nYparam === void 0) { nYparam = 0; }
        }
        return Point;
    }());
    Vectors.Point = Point;
    var cPoint = (function (_super) {
        __extends(cPoint, _super);
        function cPoint(nXparam, nYparam, nZparam) {
            if (nXparam === void 0) { nXparam = 0; }
            if (nYparam === void 0) { nYparam = 0; }
            if (nZparam === void 0) { nZparam = 0; }
            var _this = _super.call(this, nXparam, nYparam) || this;
            _this.bCurrentlyValid = true;
            _this.bHasInterSection = false;
            _this.z = 0;
            _this.x = nXparam;
            _this.y = nYparam;
            _this.z = nZparam;
            _this.bCurrentlyValid = true;
            return _this;
        }
        cPoint.prototype.ToNormalPoint = function () {
            var ptReturnValue = new Point();
            if (this.bCurrentlyValid) {
                ptReturnValue.x = this.x;
                ptReturnValue.y = this.y;
                return ptReturnValue;
            }
            ptReturnValue = null;
            return ptReturnValue;
        };
        cPoint.prototype.tostring = function () {
            var strReturnValue = "(" + this.x + "," + this.y;
            if (this.z != 0)
                strReturnValue += ", " + this.z + ")";
            else
                strReturnValue += ")";
            if (!this.bCurrentlyValid)
                strReturnValue += " INVALID";
            return strReturnValue;
        };
        return cPoint;
    }(Point));
    Vectors.cPoint = cPoint;
})(Vectors || (Vectors = {}));
